datashape/discovery.py

```diff
@@ -405,3 +405,19 @@ def descendents(d, x):
         children -= desc
         desc.update(children)
     return desc
+
+
+Mock = None
+try:
+    from unittest.mock import Mock
+except ImportError:
+    try:
+        from mock import Mock
+    except ImportError:
+        pass
+
+if Mock is not None:
+    @dispatch(Mock)
```

Comment:
it does; however, it does this:

```python
    if hasattr(o, 'shape') and hasattr(o, 'dtype'):                             |
        return from_numpy(o.shape, o.dtype)
```
and mock does have these attributes but they are mock's and then this is where the segfault occurs.